### ISSUE TITLE
GrayscaleRenderOptions.FromDataset crashes from 5.1.3

### ISSUE THREAD

**Issue by tuket** [fo-dicom/fo-dicom#1828]
In my application I use "GrayscaleRenderOptions.FromDataset" to retrieve the options:

```cs
var options = GrayscaleRenderOptions.FromDataset(dataset, frameI);
```

My application uses fo.dicom 5.1.2. I wanted to update to 5.1.3 and I relilized that the API has changed so my code wasn't compiling anymore. I need to provide a "frame" parameter now.
So I changed my code to provide a "frame".

```cs
for (int frameI = 0; frameI < numFrames; frameI++)
{
    var options = GrayscaleRenderOptions.FromDataset(dataset, frameI);
    var modalityLUT = new ModalityRescaleLUT(options);
    var fullRangePixelData = PixelDataFactory.Create(uncompressedDicomPixeldata, frameI);
    fullRangePixelData.Render(modalityLUT, buffer);
    buffer.CopyTo(texture3Darray, arrayPosition);
    arrayPosition += buffer.Length;
}
```

Now it compiles... but it crashes.

This [documentation](https://github.com/fo-dicom/fo-dicom/blob/development/Documentation/v5/usage/lut_application.md/) still references to the function that has only one parameter, so I can't find an example of how to use new API.

**Comment by tuket** [comment]
In this [documentation](https://fo-dicom.github.io/dev/v5/api/FellowOakDicom.Imaging.DicomImage.html) it suggests using a negative number, but it didn't work for me.

Upon inspecting the code, I got the idea to pass in a huge number, and that seems to work. I have no idea what I'm doing.

**Comment by mrbean-bremen** [comment]
Can you show the traceback from that crash? Also it would help to know the` SOPClassUID` and the relevant pixel tags (`BitsAllocated`/`BitsStored`, `PhotometricInterpretation`, `NumberOfFrames`).

At a first glance, this is certainly a documentation issue, and probably also a bug in the handling of the frame number. It would probably also help to provide a sensible default for the `frame` parameter used for single frame images.

**Comment by gofal** [comment]
Agree, the documentation is missing some parts. The frame parameter will only be evaluated if the image is a multiframe image, and there also only, if there are PerFrameFunctionalGroupSequences.
So the code should not crash on all images, but only on certain images, can you verify that?

**Comment by gofal** [comment]
What kind of image did you use? the method ready GrayscaleRenderOptions. internally this method is called after the Render-method checked, that PhotometricInterpretation is eigher MONOCHROME1 or MONOCHROME2 and that the values WindowWidth and WindowCenter are present in the DicomDataset.
So of course, if you have some RGB image, then this method might fail, but then it fails for good reason.

**Comment by tuket** [comment]
Hi, this was a long time ago and I don't remember. The only think I can tell is that it wasn't RGB.
